### PR TITLE
Report potential canon tip via the `getnetworkgraph` rpc

### PR DESCRIPTION
**Pull request by niklaslong**
The crawler currently only reports potential fork clusters which don't include the canon chain candidates. This PR adds the crawler's best guess at the canon tip (3+ members and heighest height in the graph) to the values exposed via rpc.

### CODE REVIEW

File: rpc/documentation/public_endpoints/getnetworkgraph.md [hunk 1]
```diff
@@ -13,6 +13,8 @@ None
 | `density`                            | f64        | The density of the known network                                                          |
 | `algebraic_connectivity`             | f64        | The algebraic connectivity, aka the fiedler eigenvalue of the known network               |
 | `degree_centrality_delta`            | f64        | The difference between the largest and the smallest peer count in the network             |
+| `potential_tip`                      | Option     | The crawler's best guess at the canon tip                                                 |
```
Comment (ljedrz) [inline rpc/documentation/public_endpoints/getnetworkgraph.md:16]:
```suggestion
| `potential_tip`                      | Option     | The crawler's best guess at the canon block height                                        |
```

File: rpc/src/rpc_impl.rs [hunk 2]
```diff
@@ -493,8 +493,10 @@ impl RpcFunctions for RpcImpl {
             })
             .collect();
 
-        let potential_forks = known_network
-            .potential_forks()
+        let (potential_tip, potential_forks) = known_network.potential_forks();
+        let potential_tip = potential_tip.map(|(height, members)| PotentialFork { height, members });
```
Comment (ljedrz) [inline rpc/src/rpc_impl.rs:497]: maybe `PotentialFork` could have its name changed to `NodeCluster`, now that it applies to both forks and the possible canon cluster?

File: network/src/topology.rs [hunk 3]
```diff
@@ -218,8 +218,9 @@ impl KnownNetwork {
         self.nodes.read().clone()
     }
 
-    /// Returns a map of the potential forks.
-    pub fn potential_forks(&self) -> HashMap<u32, Vec<SocketAddr>> {
+    #[allow(clippy::type_complexity)]
+    /// Returns the canon tip height and members, and a map of the potential forks.
+    pub fn potential_forks(&self) -> (Option<(u32, Vec<SocketAddr>)>, HashMap<u32, Vec<SocketAddr>>) {
```
Comment (ljedrz) [inline network/src/topology.rs:223]: maybe we can reuse the `PotentialFork` (or `NodeCluster`, as suggested later on) here for a simpler return type?

**Review by ljedrz — APPROVED** [review]
LGTM with some ergonomics suggestions

File: network/src/topology.rs (hunk 3 above)
Comment (niklaslong) [reply network/src/topology.rs:223]: I wanted to avoid a dependency on the RPC types but I can certainly wrap these in a new type?

Comment (ljedrz) [reply network/src/topology.rs:223]: maybe the RPC type could instead reside here?

File: rpc/documentation/public_endpoints/getnetworkgraph.md (hunk 1 above)
Comment (niklaslong) [reply rpc/documentation/public_endpoints/getnetworkgraph.md:16]: This doesn't only contain the block height but also the member nodes, I'll make this explicit though 👍

File: network/src/topology.rs (hunk 3 above)
Comment (niklaslong) [reply network/src/topology.rs:223]: Fine by me 👍

**Pull request merged**